.github/workflows/lintdiff.yaml

```diff
@@ -0,0 +1,68 @@
+name: LintDiff (preview)
+
+on: pull_request
+
+permissions:
+  contents: read
+
+jobs:
+  lintdiff:
+    name: LintDiff (preview)
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout eng
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            eng/
+            .github/
+
+      - name: Checkout 'after' state
```

Comment:
We might be able to optimize the checkouts to only the folders with files changed in the PR diff.

Reply:
Added in https://github.com/Azure/azure-rest-api-specs/issues/33302